src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/health/YarnConnectionHealthIndicator.scala

```diff
@@ -0,0 +1,45 @@
+
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.hyperdrive.trigger.api.rest.health
+
+import org.springframework.boot.actuate.health.{Health, HealthIndicator}
+import org.springframework.stereotype.Component
+import za.co.absa.hyperdrive.trigger.scheduler.utilities.{HealthConfig, SparkExecutorConfig}
+
+import java.net.{HttpURLConnection, URL}
+
+@Component
+class YarnConnectionHealthIndicator extends HealthIndicator {
+  val successCode = 200
+
+  override protected def health(): Health = {
+    val yarnBaseUrl = SparkExecutorConfig.getHadoopResourceManagerUrlBase.stripSuffix("/")
```

Comment:
I would wrap every health indicator to try-catch block.
For example my Yarn is down and health check call return 500 with logged exception: Caused by: java.net.ConnectException: Connection refused (Connection refused)

Reply:
Absolutely. Fixed it